stdlib/public/TensorFlowCore/Tensor.swift

```diff
@@ -685,8 +667,13 @@ internal extension Tensor where Scalar : TensorFlowFloatingPoint {
   ) -> (Tensor, (Tensor) -> (Tensor, Tensor)) {
     return (lhs - rhs, {
       [lhsShape = lhs.shapeTensor, rhsShape = rhs.shapeTensor] v in
-      (v.unbroadcasted(toShape: lhsShape),
-        Raw.neg(v).unbroadcasted(toShape: rhsShape))
+      if lhsShape == rhsShape {
+        return (v, Raw.neg(v))
```

Comment:
Why not use `-(_:)` instead of `Raw.neg(_:)`?

Reply:
Because `-` has been moved to `TensorFlow`. Ideally we'd want all of that to move there, but I need the `Differentiable` conformance for `Tensor` in `TensorFlowCore`, in order to allow `Tensor.init(_ value: Scalar)` and `Tensor.scalarized()` to be differentiable (these functions use `_TFTensorFromScalar` and `_TFGetScalarOrDie` which we do not expose outside of `TensorFlowCore`.

Reply:
Why is there a VJP in TensorFlowCore? I think that the Tensor type should be defined entirely in the TensorFlow module.

Reply:
Feel free to remove all the scalar promotion logic such as “_TFGetScalarOrDie” if that’s blocking you!

Reply:
That's what I wanted to do ideally, but wasn't sure if that was ok. That's a great relief actually. I'll try to move more things over and let you know to take a look and confirm it's all good. I'll try to do that within the next hour or two.

Reply:
I think one issue I run into was that the GPE tests were failing when I did that before.

Reply:
Should I also remove all the `_TFTensorFromScalars` functions? I was worried that this would affect the partitioning phase, but also wasn't sure if this is relevant anymore.

Reply:
@rxwei In this case, if we don't care about the partitioning phase anymore, can I also remove all uses of `@inline(__always)` from the new `TensorFlow` module? This will cause failures across multiple tests related to GPE.

Reply:
I am currently in the process of moving the rest of `Tensor.swift` to `swift-apis` but ran into the following issues:
- `_TFToAccelerator`, `_TFToHost`, and `_TFTensorFromScalars` all seem to be tied to the compiler through the partitioning phase (and the `@_silgen_name` attribute). Should I leave these here or move them to `swift-apis`? If I leave them here, can I make them `public` so we can use them from `swift-apis`? Otherwise I cannot move all of `Tensor.swift` there, for example.

Reply:
`_TFToAccelerator`, `_TFToHost`, `_TFTensorFromScalars`, `_TFGetScalarOrDie` are all currently unused dead code. If there are related tests that depend on them, you can keep them in the `TensorFlowCore` module as public underscored functions, and continue to use them in the `TensorFlow` module.

Reply:
Is there a way to replace uses of `#tfop` with `__shapes` and `__device` arguments with invocations of `Raw` module functions? I see `__shapes` and `__device` only used with the `Identity` op but I'm not sure what effect they have.

Reply:
There's not. Feel free to remove those arguments.

`__shapes:` was introduced early on as a workaround for XLA/TPU. 
`__device:` is for device partitioning, but that code path is not being exercised either.

Reply:
Ok cool. In that case, `Tensor.toAccelerator(shape:)` and `Tensor.toHost(shape:)` become equivalent to `Tensor.toAccelerator()` and `Tensor.toHost()` and can be removed. More generally though, if `Tensor.toAccelerator()` and `Tensor.toHost()` are not used anymore, would you mind if I remove them and also remove all relevant tests? This PR may be a good place to do that since we can clean up some unused features as we're moving them to `swift-apis`.

Reply:
@rxwei while I've been making these changes I'm starting to doubt whether we actually need `TensorFlowCore` or not. I'll leave for now, containing all the deprecated things I cannot move to `swift-apis`, but other than that I feel it may be redundant.

Reply:
Nevermind, I decided to remove it completely and move everything to `swift-apis`. I believe this makes more sense and let's us separate the compiler modifications entirely from `stdlib` which should make it easier to develop and test separately. Is it ok if I remove tests that depend on partitioning (e.g., the send/receive tests)?

Reply:
Let’s hold off on that move for now and make sure TensorHandle and CompilerRuntime related things still stay in stdlib. GPE tests still need to be run, and runtime changes need to be tested on GPU using apple/swift CI.

Reply:
Ok sure I'll do that. I'll have to comment out a couple minor things like the `dumpTensorContent` function in `CompilerRuntime.swift` that make use of either tensors or shaped arrays. I'll push the changes in a few minutes so you can take a look.

Reply:
That sounds good!